x/incentives/keeper/gauge_test.go

```diff
@@ -426,3 +426,152 @@ func (s *KeeperTestSuite) TestAddToGaugeRewards() {
 		})
 	}
 }
+
+// TestCreateGauge_NoLockGauges tests the CreateGauge function
+// specifically focusing on the no lock gauge type and test cases around it.
+// It tests the following:
+// - For no lock gauges, a CL pool id must be given and then pool must exist
+// - For no lock gauges, the denom must be set either to NoLockInternalGaugeDenom(<pool id>)
+// or be unset. If set to anything other than the internal prefix, fails with error.
+// Assumes that the gauge was created externally (via MsgCreateGauge) if the denom is unset and overwrites it
+// with NoLockExternalGaugeDenom(<pool id>)
+// - Otherwise, the given pool id must be zero. Errors if not.
+func (s *KeeperTestSuite) TestCreateGauge_NoLockGauges() {
+	const (
+		zeroPoolId         = uint64(0)
+		balancerPoolId     = uint64(1)
+		concentratedPoolId = uint64(2)
+		invalidPool        = uint64(3)
+		// 3 are created for balancer pool and 1 for CL.
+		// As a result, the next gauge id should be 5.
+		defaultExpectedGaugeId = uint64(5)
+
+		defaultIsPerpetualParam = false
+
+		defaultNumEpochPaidOver = uint64(10)
+	)
+
+	var (
+		defaultCoins = sdk.NewCoins(
+			sdk.NewCoin("uosmo", sdk.NewInt(100000)),
+			sdk.NewCoin("atom", sdk.NewInt(99999)),
+		)
+
+		defaultTime = time.Unix(0, 0)
+	)
+	testCases := []struct {
+		name    string
+		distrTo lockuptypes.QueryCondition
+		poolId  uint64
+
+		expectedGaugeId  uint64
+		expectedDenomSet string
+		expectErr        bool
+	}{
+		{
+			name: "create valid no lock gauge with CL pool (no denom set)",
+			distrTo: lockuptypes.QueryCondition{
+				LockQueryType: lockuptypes.NoLock,
+				// Note: this assumes the gauge is external
+				Denom: "",
+			},
+			poolId: concentratedPoolId,
+
+			expectedGaugeId:  defaultExpectedGaugeId,
+			expectedDenomSet: types.NoLockExternalGaugeDenom(concentratedPoolId),
+			expectErr:        false,
+		},
+		{
+			name: "create valid no lock gauge with CL pool (denom set to uosmo)",
+			distrTo: lockuptypes.QueryCondition{
+				LockQueryType: lockuptypes.NoLock,
+				// Note: this assumes the gauge is internal
+				Denom: types.NoLockInternalGaugeDenom(concentratedPoolId),
+			},
+			poolId: concentratedPoolId,
+
+			expectedGaugeId:  defaultExpectedGaugeId,
+			expectedDenomSet: types.NoLockInternalGaugeDenom(concentratedPoolId),
+			expectErr:        false,
+		},
+		{
+			name: "fail to create gauge because invalid denom is set",
+			distrTo: lockuptypes.QueryCondition{
+				LockQueryType: lockuptypes.NoLock,
+				// Note: this is invalid for NoLock gauges
+				Denom: "uosmo",
+			},
+			poolId: concentratedPoolId,
+
+			expectErr: true,
+		},
+		{
+			name: "fail to create no lock gauge with balancer pool",
+			distrTo: lockuptypes.QueryCondition{
+				LockQueryType: lockuptypes.NoLock,
+			},
```

Comment:
Does denom default to a blank string or do we still need to be explicit with this here?

Reply:
Defaults to blank string